core/src/consensus/tendermint/engine.rs

```diff
@@ -515,22 +515,22 @@ fn final_rewards(intermediate_reward: u64, number_of_signatures: u64, number_of_
         );
         (0, 0)
     };
-    let numerator = i128::from(intermediate_reward)
-        * (a * i128::from(number_of_signatures) + b * i128::from(number_of_blocks_in_term));
+    let numerator =
+        i128::from(intermediate_reward) * (a * (number_of_signatures as i128) + b * (number_of_blocks_in_term as i128));
```

Comment:
You'd better prefer to use `from` instead of `as`. Using `as` coerces into type conversion even it's not safe.

Reply:
`From<usize>` is not implemented for `i128`. Do you have any idea for solving this?

Reply:
You can use `TryFrom` with `unwarp`. `TryFrom<usize> for i128` is safe to unwrap because it always returns `Ok`

Reply:
Hmm. Let me try it.